ext/reflection/php_reflection.c

```diff
@@ -1443,7 +1489,8 @@ static void reflection_method_factory(zend_class_entry *ce, zend_function *metho
 	intern->ref_type = REF_TYPE_FUNCTION;
 	intern->ce = ce;
 	if (closure_object) {
-		ZVAL_OBJ_COPY(&intern->obj, Z_OBJ_P(closure_object));
+		intern->ce = zend_ce_closure;
```

Comment:
TODO: This looks wrong, and I need to add a test with a Closure bound within a method.